### PR TITLE
fix(docs-infra): add deprecated-api-item class to remaining deprecated items

### LINKED ISSUE #31455: Mark deprecated API properties with strike-out font
# 📚 Docs or angular.io bug report

### Description

The API docs show many deprecated elements with strike-out font. Within classes and interfaces, deprecated methods are correctly struck out, but deprecated *properties* are not.
## 🔬 Minimal Reproduction

See `https://angular.io/api/router/NavigationExtras#preserveQueryParams` (deprecated in favor of `queryParamsHandling` since v4.

### What's the affected URL?**
`https://angular.io/api/router/NavigationExtras#preserveQueryParams` 
(might be more)

### Expected vs Actual Behavior**
The deprecated property should appear in strike-out font.

### PR DESCRIPTION
**Pull request by ajitsinghkaler**
Fixes #31455

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
It adds strikethrough to deprecate items

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [x] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Deprecated items are not striked through

Issue Number: #31455

## What is the new behavior?
Deprecated Items are stroked through

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by ajitsinghkaler** [comment]
@petebacondarwin I have made the rest of the changes you can have a look

File: aio/tools/transforms/templates/api/lib/memberHelpers.html [hunk 1]
```diff
@@ -12,15 +12,15 @@
...
+  <a class="code-anchor  {% if member.deprecated %} deprecated-api-item{% endif %}" href="{$ doc.path $}#{$ doc.constructorDoc.anchor | urlencode $}">{$ renderMemberSyntax(doc.constructorDoc, 1) $}</a>{% endif -%}
   {% for member in doc.properties %}{% if not member.internal %}
-  <a class="code-anchor {% if member.deprecated %} deprecated-api-item{% endif %}" href="{$ doc.path $}#{$ member.anchor | urlencode $}">{$ renderMemberSyntax(member, 1) $}</a>{% endif %}{% endfor -%}
+  <a class="code-anchor  {% if member.deprecated %} deprecated-api-item{% endif %}" href="{$ doc.path $}#{$ member.anchor | urlencode $}">{$ renderMemberSyntax(member, 1) $}</a>{% endif %}{% endfor -%}
   {% for member in doc.methods %}{% if not member.internal %}
-  <a class="code-anchor {% if member.deprecated %} deprecated-api-item{% endif %}" href="{$ doc.path $}#{$ member.anchor | urlencode $}">{$ renderMemberSyntax(member, 1) $}</a>{% endif %}{% endfor -%}
+  <a class="code-anchor  {% if member.deprecated %} deprecated-api-item{% endif %}" href="{$ doc.path $}#{$ member.anchor | urlencode $}">{$ renderMemberSyntax(member, 1) $}</a>{% endif %}{% endfor -%}
```
Comment (gkalpak) [inline aio/tools/transforms/templates/api/lib/memberHelpers.html]: Why these extra spaces? In fact, I don't think there should be any space before `{% if`.

File: aio/tools/transforms/templates/api/decorator.template.html [hunk 2]
```diff
@@ -26,7 +26,7 @@ <h3>{$ option.name $}</h3>
       </tr>
       <tr>
         <td>
-          <code-example language="ts" hideCopy="true" class="no-box api-heading">
+          <code-example language="ts" hideCopy="true" class="no-box api-heading {% if option.deprecated %} deprecated-api-item{% endif %}">
```
Comment (gkalpak) [inline aio/tools/transforms/templates/api/decorator.template.html]: I don't think there should be a whitespace before `{% if`. If it is used like this throughout the code and want to keep it for consistency, then we don't need a space before `deprecated-api`.

(The same applies to other files as well.)

**Review by gkalpak — APPROVED** [review]
Looks reasonable. Do we know of any examples where these changes actually make a difference (i.e. deprecated stuff that would not show up correctly without these changes)?

**Comment by gkalpak** [comment]
BTW, the commit message type should be `docs-infra`, not `docs`.

**Comment by ajitsinghkaler** [comment]
@gkalpak I'm attaching images for reference where it removes deprecated items.
Angular - Testability (3)
Angular - Testability (2)
Angular - Testability (1)
Angular - Testability

**Comment by ajitsinghkaler** [comment]
@gkpalak I've made the change please have a look at them.

File: aio/tools/transforms/templates/api/lib/memberHelpers.html [hunk 3]
```diff
@@ -46,7 +46,7 @@
     {$ overload.shortDescription | marked $}
   </div>{% endif %}
 
-  <code-example language="ts" hideCopy="true" class="no-box api-heading">{$ renderMemberSyntax(overload) $}</code-example>
+  <code-example language="ts" hideCopy="true" class="no-box api-heading {% if overload.deprecated %} deprecated-api-item{% endif %}">{$ renderMemberSyntax(overload) $}</code-example>
```
Comment (gkalpak) [inline aio/tools/transforms/templates/api/lib/memberHelpers.html]: Missed an extra space before `{% if` here :smiley:

**Review by gkalpak — APPROVED** [review]
Thx, @ajitsinghkaler! The link and screenshots are really helpful 👌

I wonder if it might be better to strikethrough the method-table header (instead of the signature): 
deprecated-method

File: aio/tools/transforms/templates/api/lib/memberHelpers.html (hunk 3 above)
Comment (ajitsinghkaler) [reply aio/tools/transforms/templates/api/lib/memberHelpers.html]: I've made the change

**Comment by ajitsinghkaler** [comment]
@gkalpak So, should I add strikethrough to the headings or it is just fine.

**Comment by ajitsinghkaler** [comment]
@gkalpak I also don't know why these tests are failing

**Comment by gkalpak** [comment]
The more I think about it, the more I don't like striking through the signature.
So, unless someone objects, I believe that (if we want to strike through something) we should be striking through the headers.

@petebacondarwin, wdyt?

**Comment by ajitsinghkaler** [comment]
@gkalpak okay, thanks for all the guidance.  I'll put strkethrough in the heading once @petebacondarwin confirms that we should do it.

**Comment by petebacondarwin** [comment]
One problem, I think.
The signatures are per overload and there might be scenarios where one overload is deprecated while others are not…

**Comment by gkalpak** [comment]
Agh...good point. OK, then. Let's keep the strike-through on the signature for now. We can revisit later.

**Review by gkalpak — APPROVED** [review]

**Pull request closed without merging**